### PR TITLE
Use discovery_bootif for primary interface detection

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/lib/puppet_fact_parser_extensions.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module PuppetFactParserExtensions
+  extend ActiveSupport::Concern
+
+  included do
+    alias_method_chain :primary_interface, :discovery_fact
+    alias_method_chain :ip, :discovery_fact
+  end
+
+  # we prefere discovery_bootif fact to choose right primary interface (interface used to boot the image)
```
Comment (lzap) [inline app/lib/puppet_fact_parser_extensions.rb]: `prefere` typo

**Comment by lzap** [comment]
LGTM @GregSutcliffe ?

Since this is a big change that can possibly break things up, can you please add a issue and add reference to the commit message (as well as downstream BZ number in the issue)? Thanks. Going to test this.

**Comment by ares** [comment]
Added both links.

**Comment by GregSutcliffe** [comment]
This seems sane on a first read, how did the testing go @lzap ?

**Comment by lzap** [comment]
Ok my discovery works fine, it starts Anaconda and configure network.

**Comment by lzap** [comment]
RTM

**Comment by ares** [comment]
I hope RTM does not mean read the manual here :-)

**Comment by lzap** [comment]
It does not. :) Thanks @ares merged as 6bf4687

**Pull request closed without merging**